packages/grid/x-data-grid-premium/src/hooks/features/clipboard/useGridClipboardImport.ts

```diff
@@ -0,0 +1,426 @@
+    const rowsToUpdate = this.rowsToUpdate;
+    const rowIdsToUpdate = Object.keys(rowsToUpdate);
+
+    if (rowIdsToUpdate.length === 0) {
+      return;
+    }
+
+    apiRef.current.publishEvent('clipboardPasteStart');
+
+    const handleRowUpdate = async (rowId: GridRowId) => {
+      const newRow = rowsToUpdate[rowId];
+
+      if (typeof processRowUpdate === 'function') {
+        const handleError = (errorThrown: any) => {
+          if (onProcessRowUpdateError) {
+            onProcessRowUpdateError(errorThrown);
+          } else {
+            missingOnProcessRowUpdateErrorWarning();
+          }
+        };
+
+        try {
+          const oldRow = apiRef.current.getRow(rowId);
+          const finalRowUpdate = await processRowUpdate(newRow, oldRow);
+          this.updateRow(finalRowUpdate);
+        } catch (error) {
+          handleError(error);
+        }
+      } else {
+        this.updateRow(newRow);
+      }
+    };
+
+    const promises = rowIdsToUpdate.map((rowId) => {
+      // Wrap in promise that always resolves to avoid Promise.all from stopping on first error.
+      // This is to avoid using `Promise.allSettled` that has worse browser support.
+      return new Promise((resolve) => {
+        handleRowUpdate(rowId).then(resolve).catch(resolve);
+      });
+    });
+    Promise.all(promises).then(() => {
+      this.rowsToUpdate = {};
+      apiRef.current.publishEvent('clipboardPasteEnd');
+    });
+  }
+}
+
+export const useGridClipboardImport = (
+  apiRef: React.MutableRefObject<GridPrivateApiPremium>,
+  props: Pick<
+    DataGridPremiumProcessedProps,
+    | 'pagination'
+    | 'paginationMode'
+    | 'processRowUpdate'
+    | 'onProcessRowUpdateError'
+    | 'getRowId'
+    | 'onClipboardPasteStart'
+    | 'onClipboardPasteEnd'
+    | 'experimentalFeatures'
+  >,
+): void => {
+  const processRowUpdate = props.processRowUpdate;
+  const onProcessRowUpdateError = props.onProcessRowUpdateError;
+  const getRowId = props.getRowId;
+  const enableClipboardPaste = props.experimentalFeatures?.clipboardPaste ?? false;
+  const rootEl = apiRef.current.rootElementRef?.current;
+
+  const handlePaste = React.useCallback<GridEventListener<'cellKeyDown'>>(
+    async (params, event) => {
+      if (!enableClipboardPaste) {
+        return;
+      }
+      const isModifierKeyPressed = event.ctrlKey || event.metaKey || event.altKey;
+      if (String.fromCharCode(event.keyCode) !== 'V' || !isModifierKeyPressed) {
+        return;
+      }
+
+      const focusedCell = gridFocusCellSelector(apiRef);
+      if (focusedCell !== null) {
+        const cellMode = apiRef.current.getCellMode(focusedCell.id, focusedCell.field);
+        if (cellMode === 'edit') {
+          // Do not paste data when the cell is in edit mode
+          return;
+        }
+      }
+
+      if (!rootEl) {
+        return;
+      }
+
+      // Do not enter cell edit mode on paste
+      event.defaultMuiPrevented = true;
```

Comment:
> I still think that the editing feature should be fixed because, even if the clipboard import is not enabled, pressing Ctrl+v is putting a "v" character and entering the edit mode.

This is actually the current behavior in the latest version: https://codesandbox.io/s/elegant-surf-pc6sgw?file=/demo.js
I don't think "v" character was ever intended there, but as far as I remember Ctrl+v should enter the edit mode.

If the clipboard import is not enabled, the grid should handle Ctrl+V the way it handled it before.
This means that in the editing hook, the following conditions will be necessary:
```tsx
if (
  event.code === 'KeyV' &&
  isModifierKeyPressed &&
  signature === GridSignature.DataGridPremium &&
  // @ts-ignore `clipboardPaste` flag only exists in Premium package 
  experimentalFeatures?.clipboardPaste
) {
  return;
}
```

Do you think it's the right way of doing it? I don't mind `@ts-ignore` there, but I was looking for a way to avoid it if necessary - that is why I used `defaultMuiPrevented`

Reply:
Added `canStartEditing` processor and used it instead of `defaultMuiPrevented`